test/span.js

```diff
@@ -185,31 +184,41 @@ describe('span should', () => {
         assert.isOk(unnormalizedKey in Span._getBaggageHeaderCache());
     });
 
-    describe('with deferred sampling', () => {
```

Comment:
Most test cases that this test covered are covered in the `with upsampling` test.
The `should not pass deferred sampling flag to child spans` no longer applies to upsampling because we don't have a special flag.